Emby.Server.Implementations/LiveTv/TunerHosts/HdHomerun/HdHomerunManager.cs

```diff
@@ -111,7 +111,7 @@ public void Dispose()
 
         public async Task<bool> CheckTunerAvailability(IPAddress remoteIp, int tuner, CancellationToken cancellationToken)
         {
-            using (var client = new TcpClient(new IPEndPoint(remoteIp, HdHomeRunPort)))
+            using (var client = new TcpClient(remoteIp.ToString(), HdHomeRunPort))
```

Comment:
The changes to this file shouldn't have any effect. You're just using a different method to connect to the endpoint

Reply:
These constructors are *not* the same per the documentation:

TcpClient(IPEndPoint) - Initializes a new instance of the TcpClient class and *binds it to the specified local endpoint*.
TcpClient(String, Int32) - Initializes a new instance of the TcpClient class and *connects to the specified port on the specified host*.

See here: https://docs.microsoft.com/en-us/dotnet/api/system.net.sockets.tcpclient?view=netcore-3.1

Reply:
You can also confirm this here as the .NET Core source code is more clear than .NET Framework: https://github.com/dotnet/runtime/blob/master/src/libraries/System.Net.Sockets/src/System/Net/Sockets/TCPClient.cs#L48

As you can see, `TcpClient(IPEndPoint localEP)` ultimately calls `_clientSocket.Bind(localEP);`, thus attempting to bind itself to the IPEndPoint specified. Given that the Jellyfin source is passing in the remort IP, it's trying to bind locally to the remote IP, which is never going to work. `TcpClient(string hostname, int port)` on the other hand calls `Connect(string hostname, int port)` which then calls `_clientSocket.Connect(address, port);` (via `socket`) which is what we want.

Reply:
Oops - you are entirely right!

Had a look at the source code - wouldn't it still work if a new TcpClient() was created (no params) then to call Connect(IPEndPoint()) used instead?
From the source code, this method looks like it would bypass all the reverse dns stuff, but still connects as required.

Reply:
Looks like that would work too.

Reply:
Pushed an update to use TcpClient.Connect(). Also confirmed it's still working as expected after the change.